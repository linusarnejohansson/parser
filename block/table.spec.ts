import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("block.table", () => {
	it("simple", () => {
		const result = parse(new Source(IO.StringReader.create("| Header A | Header B |\n|----------|----------|\n| value A  | value B  |\nTable Caption."), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("complex", () => {
		const result = parse(new Source(IO.StringReader.create(`
| Property          |   | Description                       |
|-------------------|---|-----------------------------------|
| name              | r | Name of the site.                 |
| url               | r | URL of the site.                  |
| terms             | o | The terms and conditions.         |
A selection of relevant properties required for creating new merchants. (r - required, o - optional)
		`), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
