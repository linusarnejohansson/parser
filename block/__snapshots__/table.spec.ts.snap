// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.table complex 1`] = `
Array [
  Object {
    "class": "Block.EmptyLine",
  },
  Object {
    "alignments": Array [
      "",
      "",
      "",
    ],
    "class": "Block.Table",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "A selection of relevant properties required for creating new merchants. (r - required, o - optional)
",
      },
      Object {
        "class": "Inline.Text",
        "value": "		",
      },
    ],
    "rows": Array [
      Object {
        "class": "Block.TableRow",
        "content": Array [
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "Property          ",
              },
            ],
            "header": true,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "   ",
              },
            ],
            "header": true,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " Description                       ",
              },
            ],
            "header": true,
          },
        ],
      },
      Object {
        "class": "Block.TableRow",
        "content": Array [
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "name              ",
              },
            ],
            "header": false,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " r ",
              },
            ],
            "header": false,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " Name of the site.                 ",
              },
            ],
            "header": false,
          },
        ],
      },
      Object {
        "class": "Block.TableRow",
        "content": Array [
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "url               ",
              },
            ],
            "header": false,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " r ",
              },
            ],
            "header": false,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " URL of the site.                  ",
              },
            ],
            "header": false,
          },
        ],
      },
      Object {
        "class": "Block.TableRow",
        "content": Array [
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "terms             ",
              },
            ],
            "header": false,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " o ",
              },
            ],
            "header": false,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " The terms and conditions.         ",
              },
            ],
            "header": false,
          },
        ],
      },
    ],
  },
]
`;

exports[`block.table simple 1`] = `
Array [
  Object {
    "alignments": Array [
      "",
      "",
    ],
    "class": "Block.Table",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "Table Caption.",
      },
    ],
    "rows": Array [
      Object {
        "class": "Block.TableRow",
        "content": Array [
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "Header A ",
              },
            ],
            "header": true,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " Header B ",
              },
            ],
            "header": true,
          },
        ],
      },
      Object {
        "class": "Block.TableRow",
        "content": Array [
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "value A  ",
              },
            ],
            "header": false,
          },
          Object {
            "class": "Block.TableCell",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": " value B  ",
              },
            ],
            "header": false,
          },
        ],
      },
    ],
  },
]
`;
